### PR TITLE
Fix CI and Regressions relating to strings

### PR DESCRIPTION
**Pull request by dwightguth**
Ready for review.

### CODE REVIEW

**Comment by dwightguth** [comment]
Jenkins: test this please

**Comment by dwightguth** [comment]
Jenkins: test this please

**Comment by dwightguth** [comment]
Jenkins: test this please

**Comment by dwightguth** [comment]
Jenkins: test this please

File: libsolidity/codegen/IeleCompiler.cpp [hunk 1]
```diff
@@ -5017,10 +5021,18 @@ bool IeleCompiler::shouldCopyStorageToStorage(const IeleLValue *To,
...
+bool IeleCompiler::shouldCopyMemoryToStorage(const Type &ToType, const IeleLValue *To,
                                              const Type &From) const {
   return dynamic_cast<const ReadOnlyLValue *>(To) &&
-         From.dataStoredIn(DataLocation::Memory);
+         From.dataStoredIn(DataLocation::Memory) &&
+         ToType.dataStoredIn(DataLocation::Storage);
+}
+
+bool IeleCompiler::shouldCopyMemoryToMemory(const Type &ToType, const IeleLValue *To,
+                                             const Type &From) const {
```
Comment (dfilaretti) [inline libsolidity/codegen/IeleCompiler.cpp]: Having the two types named `ToType` and `From`, and the actual value named `To` can be a little confusing. Why not using e.g.:

```
bool IeleCompiler::shouldCopyMemoryToMemory(const Type &ToType, const IeleLValue *Value,
+                                             const Type &FromType) const {
```

Comment (theo25) [reply libsolidity/codegen/IeleCompiler.cpp]: If you do a renaming here, I would suggest only changing `From` to `FromType`. Also you should do it in both this function and the one below.

**Comment by theo25** [comment]
The new changes also look good.

**Pull request merged**